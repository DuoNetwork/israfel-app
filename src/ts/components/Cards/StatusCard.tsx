import { Table } from 'antd';
import * as React from 'react';
import * as CST from 'ts/common/constants';
import util from 'ts/common/util';
import { IStatus } from '../../../../../israfel-common/src';
import { SCard, SCardTitle, STableWrapper } from './_styled';

const { Column } = Table;

interface IProps {
	status: IStatus[];
}

export default class StatusCard extends React.Component<IProps> {
	public render() {
		const { status } = this.props;
		const dataSource = status.map(s => ({
			key: s.tool + s.pair + s.hostname,
			[CST.TH_PROCESS]: s.tool,
			[CST.TH_PAIR]: s.pair.replace('|', '-'),
			[CST.TH_UPDATED]: util.convertUpdateTime(s.updatedAt),
			[CST.TH_HOSTNAME]: s.hostname,
			[CST.TH_INFO]: s.count
		}));
		dataSource.sort((a, b) => a.key.localeCompare(b.key));
		return (
			<SCard
				title={<SCardTitle>{CST.TH_STATUS}</SCardTitle>}
				width="100%"
				margin="20px 0 0 0"
				inlinetype="table"
			>
				<STableWrapper>
					<Table dataSource={dataSource} pagination={false}>
						{[
							CST.TH_PROCESS,
							CST.TH_PAIR,
							CST.TH_UPDATED,
							CST.TH_HOSTNAME,
							CST.TH_INFO
						].map(th => (
							<Column title={th} dataIndex={th} key={th} />
						))}
					</Table>
				</STableWrapper>
			</SCard>
		);
	}
}
