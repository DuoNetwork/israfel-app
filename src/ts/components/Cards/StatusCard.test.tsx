/**
 * @jest-environment jsdom
 */

import { shallow } from 'enzyme';
import * as React from 'react';
import StatusCard from './StatusCard';

describe('StatusCard Test', () => {
	describe('Test Snapshot', () => {
		const status = [{
			hostname: "test",
			updatedAt: 123,
			pair: "test",
			tool: "test",
			count: 123,
		}];
		it('Test Snapshot', () => {
			const wrapper = shallow(<StatusCard status={status} />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
