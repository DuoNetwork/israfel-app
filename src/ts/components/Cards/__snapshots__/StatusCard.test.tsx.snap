// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCard Test Test Snapshot Test Snapshot 1`] = `
<Styled(Card)
  inlinetype="table"
  margin="20px 0 0 0"
  title={
    <styled.div>
      Status
    </styled.div>
  }
  width="100%"
>
  <styled.div>
    <Table
      bordered={false}
      className=""
      dataSource={
        Array [
          Object {
            "Hostname": "test1",
            "Info": 123,
            "Pair": "test1",
            "Process": "test1",
            "Updated": "long time ago",
            "key": "test1test1test1",
          },
          Object {
            "Hostname": "test",
            "Info": 123,
            "Pair": "test",
            "Process": "test",
            "Updated": "long time ago",
            "key": "testtesttest",
          },
        ]
      }
      indentSize={20}
      loading={false}
      locale={Object {}}
      pagination={false}
      prefixCls="ant-table"
      rowKey="key"
      showHeader={true}
      size="default"
      useFixedHeader={false}
    >
      <Column
        dataIndex="Process"
        key="Process"
        title="Process"
      />
      <Column
        dataIndex="Pair"
        key="Pair"
        title="Pair"
      />
      <Column
        dataIndex="Updated"
        key="Updated"
        title="Updated"
      />
      <Column
        dataIndex="Hostname"
        key="Hostname"
        title="Hostname"
      />
      <Column
        dataIndex="Info"
        key="Info"
        title="Info"
      />
    </Table>
  </styled.div>
</Styled(Card)>
`;
